V9 Configuration documentation (#3278)
umbraco/UmbracoDocs: Reference/V9-Config/HealthChecks/index-v9.md

---
versionFrom: 9.0.0
meta.Title: "Umbraco Health Check Settings"
meta.Description: "Information on the health check settings section"
state: complete
verified-against: beta-3
update-links: true
---

# Health checks

The health checks section allows you to disable certain health checks, and configure your own custom notification methods, that will automatically run the health checks every so often, and notify you if any health checks fails.

An example of a HealthChecks settings can look something like this:

```json
"Umbraco": {
  "CMS": {
    "HealthChecks": {
      "DisabledChecks": [
        {
          "Id": "D0F7599E-9B2A-4D9E-9883-81C7EDC5616F"
        }
      ],
      "Notification": {
        "Enabled": true,
        "FirstRunTime": "5 * * * *",
        "Period": "1.00:00:00",
        "NotificationMethods": {
          "email": {
            "Enabled": true,
            "Verbosity": "Detailed",
            "FailureOnly": true,
            "Settings": {
              "RecipientEmail": "alerts@mywebsite.tld"
            }
          }
        }
      }
    }
  }
}
```

> This config will disable the macro errors check, and enable notifications to run the check and notify via email if a check fails, running the first time five minutes are the site as booted, and this once every day.

Reviewer: Not sure I understand the last part of this sentence 🤔

Author: I can see why hehe, I managed to butcher that sentence, I've tried updating it 😄

The email notification method is built in, if you want to read more about creating you own notification methods, or see a list of the ID of every built in health check, then see [Extending health checks](../../../Extending/Health-Check/index-v9.md)

But let's go through the config one by one

## Disabled checks

A list of `DisabledHealthCheckSettings` objects, each of these objects represents a disabled health check. Only the Id key needs to be present and have a value, corosponding to the GUID of the health check to disable.

There is also a `DisabledOn` key representing the date the health check was disabked and a `DisabledBy` key containing the ID of the user that disabled the health check, however these values are currently not used.

## Notification

Settings relating to running the health checks automatically and sending out notifications.

### Enabled

> Allows you to disable orenable all notifications methods, if set to false, the health checks will not automatically run.

Reviewer: ```suggestion
Allows you to disable or enable all notifications methods, if set to false, the health checks will not automatically run.
```

### First run time

> This will configure when to run the health checks for the first time, if the value is not configured the health checks will run imideately after the site boots for the first time. This value is specified as a string in crontab format, so in this example the health checks will run 5 minutes after the site is first booted.

Reviewer: ```suggestion
This will configure when you run the health checks for the first time, if the value is not configured the health checks will run immediately after the site boots for the first time. This value is specified as a string in crontab format, so in this example, the health checks will run 5 minutes after the site is first booted.
```
